resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/session/TestBug202384.java

```diff
@@ -14,96 +14,60 @@
+		workspace.save(true, getMonitor());
 	}
 
-	public void testStartWithClosedProject() {
+	public void testStartWithClosedProject() throws CoreException {
 		IWorkspace workspace = ResourcesPlugin.getWorkspace();
 		IProject project = workspace.getRoot().getProject("project");
 		assertFalse("1.0", project.isOpen());
-		try {
-			//project is closed so it is not possible to read correct encoding
-			assertNull("2.0", project.getDefaultCharset(false));
-		} catch (CoreException e) {
-			fail("3.0", e);
-		}
-		try {
-			//opening the project should initialize ProjectPreferences
-			project.open(getMonitor());
-		} catch (CoreException e) {
-			fail("4.0", e);
-		}
-		try {
-			//correct values should be available after initialization
-			assertEquals("5.0", "UTF-8", project.getDefaultCharset(false));
-		} catch (CoreException e) {
-			fail("6.0", e);
-		}
-		try {
-			workspace.save(true, getMonitor());
-		} catch (CoreException e) {
-			fail("7.0", e);
-		}
+		// project is closed so it is not possible to read correct encoding
+		assertNull("2.0", project.getDefaultCharset(false));
+		// opening the project should initialize ProjectPreferences
+		project.open(getMonitor());
+		// correct values should be available after initialization
+		assertEquals("5.0", "UTF-8", project.getDefaultCharset(false));
+		workspace.save(true, getMonitor());
 	}
 
-	public void testStartWithOpenProject() {
+	public void testStartWithOpenProject() throws CoreException {
 		IWorkspace workspace = ResourcesPlugin.getWorkspace();
 		IProject project = workspace.getRoot().getProject("project");
 		assertTrue("1.0", project.isOpen());
-		try {
-			//correct values should be available if ProjectPreferences got
-			//initialized upon creation
-			String expectedEncoding = "UTF-8";
-			// check with a timeout, in case some initialize operation is slow
-			long timeout = 10_000;
-			long start = System.currentTimeMillis();
-			while (!expectedEncoding.equals(project.getDefaultCharset(false))
-					&& System.currentTimeMillis() - start < timeout) {
-				TestUtil.dumpRunnigOrWaitingJobs(getName());
-				TestUtil.waitForJobs(getName(), 500, 1000);
-			}
-			assertEquals("2.0", expectedEncoding, project.getDefaultCharset(false));
-		} catch (CoreException e) {
-			fail("3.0", e);
-		}
-		try {
-			workspace.save(true, getMonitor());
-		} catch (CoreException e) {
-			fail("4.0", e);
+		// correct values should be available if ProjectPreferences got
+		// initialized upon creation
+		String expectedEncoding = "UTF-8";
+		// check with a timeout, in case some initialize operation is slow
+		long timeout = 10_000;
+		long start = System.currentTimeMillis();
+		while (!expectedEncoding.equals(project.getDefaultCharset(false))
+				&& System.currentTimeMillis() - start < timeout) {
+			TestUtil.dumpRunnigOrWaitingJobs(getName());
+			TestUtil.waitForJobs(getName(), 500, 1000);
 		}
+		assertEquals("2.0", expectedEncoding, project.getDefaultCharset(false));
```

Comment:
Remove the "2.0", it's useless

Reply:
I will not touch all the assertions. Of course, it would be good to get rid of the numbered messages, but that is a different issue than what this PR addresses.

Reply:
... pity, it seems like an opportunity missed. 
But it's fine, this is not a pressing issue.

Reply:
I absolutely agree that it would be nice if those messages were removed/improved. But this is either quite some effort (if you want to improve the messages) or does not give you any benefit (when only removing them). At the same time it will even further bloat up all the pull requests that are required for JUnit 4/5 migration anyway. That is something that could also be done systematically by simple find-and-replace in a dedicated PR.